homeassistant/components/zha/sensor.py

```diff
@@ -330,6 +354,13 @@ class EnumSensor(Sensor):
     _attr_device_class: SensorDeviceClass = SensorDeviceClass.ENUM
     _enum: type[enum.Enum]
 
+    def _init_from_quirks_metadata(self, entity_metadata: EntityMetadata) -> None:
+        """Init this entity from the quirks metadata."""
+        ZhaEntity._init_from_quirks_metadata(self, entity_metadata)  # pylint: disable=protected-access
+        sensor_metadata: ZCLEnumMetadata = entity_metadata.entity_metadata
+        self._attribute_name = sensor_metadata.attribute_name
```

Comment:
This will create extra state attribute that we don't see the name of. We want to review all extra attributes before they are added. There must at minimum be a map in the integration for these.

Reply:
The attribute name is at least used here for an extra state attribute:
https://github.com/home-assistant/core/blob/c2a991625d0925f420b0374a162f369991ecfa65/homeassistant/components/zha/sensor.py#L460

Reply:
There won’t be any metadata in these cases. I can add a check to them that if it is present they also throw a hard error…